spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/CertificateUtil.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.spec.bootstrap.data.util;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.commons.codec.binary.Base64;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.Inflater;
+
+public final class CertificateUtil {
+
+    private CertificateUtil() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Extracts the decoded and uncompressed bodies of the entitlement certificates.
+     *
+     * @param jsonPayload - the json that contains the encoded and compressed entitlement certificates
+     * @return the entitlement certificate bodies
+     * @throws Exception if unable to decompress the body of the certificate or parse the json
+     */
+    public static List<JsonNode> extractEntitlementCertificatesFromPayload(Object jsonPayload)
+        throws Exception {
+        if (jsonPayload == null) {
+            return new ArrayList<>();
+        }
+
+        String json = String.valueOf(jsonPayload);
+        if (json == null || json.isEmpty() || json.isBlank()) {
+            return new ArrayList<>();
+        }
+
+        // Retrieves the certificate substrings from the json.
+        List<String> certificates = new ArrayList<>();
+        int certEndIndex = 0;
+        int certStartIndex = json.indexOf("cert=", certEndIndex);
+        while (certStartIndex != -1 && certEndIndex != -1) {
```

Comment:
I don't think we will be able to do this because the json does not have quotes on the key and value. When I try to parse as a list of maps I get the following error.

```
com.fasterxml.jackson.databind.JsonMappingException: Unexpected character ('c' (code 99)): was expecting double-quote to start field name
         at [Source: (String)"[{created=2022-07-27T20:43:35+0000, ...
```

Reply:
Actually, after seeing how the ConsumerClient.fetchCertificates() is able to use the ObjectMapper to create a list of CertificateDTOs, I did the same thing and then populated a list of certificates. This is great because I really disliked parsing the String! Thanks for pointing this out!